### ISSUE TITLE
Site Deletion: console error after deletion and no confirmation

### ISSUE THREAD

**Issue by fabianapsimoes** [Automattic/wp-calypso#5084]
I'm seeing two console errors in the Site Deletion flow:

- `this.notes[0] is undefined` shows as soon as I load `/settings/general/:site`. This continues being logged in the console within regular intervals of time.
- `TypeError: ownProps.site is undefined` shows as soon as I complete the deletion flow and see the green notice.

Another thing I noticed is that there's no confirmation the site was deleted. I see a green notice saying ":site is being deleted" but nothing happens after that. Not sure if that's by design or related to the errors mentioned above.

Steps to reproduce:

- Create a new site and open your browser's console.
- Visit `/settings/general/:site` and check the console. You'll see the error about `this.notes[0]`.
- Select Delete Site at the bottom of the page, and complete the flow.
- Take another look at your console. You'll see the error about `ownProps.site`.
- After completing the flow, you'll see a green notice letting you know the site is being deleted. Nothing happens after this point.

cc @gwwar, @rralian: @stephanethomas mentioned the error about `ownProps.site` might be related to https://github.com/Automattic/wp-calypso/pull/4881 (p1461846744000049-upgrades-slack).

**Comment by lancewillett** [comment]
Related: #4444

**Comment by lancewillett** [comment]
Missing confirmation is a dupe of #158

**Comment by fabianapsimoes** [comment]
After the fix in #5089, I no longer see the error involving `ownProps.site`. Instead I see `TypeError: site is undefined` -- cc @gwwar.

The error about `this.notes[0]` is still there.

**Comment by fabianapsimoes** [comment]
Also I just observed `this.notes[0] is undefined` after signup when landing in Calypso, either in `/checkout` or `/plans/:site` (in case you got a free trial). So not specific to this flow. @lancewillett maybe worth moving this to a separate issue?

**Comment by lancewillett** [comment]
Yes, new issue would be good. Thanks!

**Comment by rralian** [comment]
I did some screen-sharing with @fabianapsimoes, looks like the `this.notes[0]` issue is related to the notifications system. I'll create a separate issue for that one. I'm looking into the site deletion issue now. We were able to recreate that issue in firefox but not chrome.

**Comment by gwwar** [comment]
> TypeError: site is undefined

@fabianapsimoes Did you happen to know what line number that's coming from?

**Comment by fabianapsimoes** [comment]
I've just tested this again, after #5190 was merged, and seems the errors are gone. 

> Another thing I noticed is that there's no confirmation the site was deleted. I see a green notice saying ":site is being deleted" but nothing happens after that. Not sure if that's by design or related to the errors mentioned above.

This is also fixed now. We're redirected to the no sites screen after the deletion.

Closing.
